flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperator.java

```diff
@@ -628,6 +644,123 @@ protected final boolean isCleanupTime(W window, long time) {
 		return time == cleanupTime(window);
 	}
 
+	public abstract class KeyedStateStoreWithWindow implements KeyedStateStore {
```

Comment:
Maybe comment that we have a base class where we can set the window so that we can once create a `MergingKeyStore` or `WindowPaneKeyStore` (depending on the window assigner) and then not care about the distinction anymore. 

I remember I suggested this but now struggled to see why there is the base class. 😉